lib/srv/app/server.go

```diff
@@ -592,29 +601,112 @@ func (s *Server) HandleConnection(conn net.Conn) {
 }
 
 func (s *Server) handleConnection(conn net.Conn) error {
+	// Make sure everything here is wrapped in the tracking read connection for monitoring.
+	ctx, cancel := context.WithCancel(s.closeContext)
+	tc, err := srv.NewTrackingReadConn(srv.TrackingReadConnConfig{
+		Conn:    conn,
+		Clock:   s.c.Clock,
+		Context: ctx,
+		Cancel:  cancel,
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
 	// Proxy sends a X.509 client certificate to pass identity information,
 	// extract it and run authorization checks on it.
-	tlsConn, user, app, err := s.getConnectionInfo(s.closeContext, conn)
+	tlsConn, user, app, err := s.getConnectionInfo(ctx, tc)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	authCtx, _, err := s.authorizeContext(context.WithValue(ctx, auth.ContextUser, user))
```

Comment:
We are. AFAICT, no -- this gets a little wonky because there are, I believe, racy reasons for wanting these both here. It should be noted this is only adding one extra auth at the start of time.